src/routes/safe/components/DropdownCurrency/index.jsx

```diff
@@ -0,0 +1,239 @@
+// @flow
+import ListItemIcon from '@material-ui/core/ListItemIcon'
+import ListItemText from '@material-ui/core/ListItemText'
+import Menu from '@material-ui/core/Menu'
+import MenuItem from '@material-ui/core/MenuItem'
+import React, { useState } from 'react'
+import style from 'currency-flags/dist/currency-flags.min.css'
+import { makeStyles, MuiThemeProvider } from '@material-ui/core/styles'
+import { useDispatch, useSelector } from 'react-redux'
+import SearchIcon from '@material-ui/icons/Search'
+import InputBase from '@material-ui/core/InputBase'
+import { DropdownListTheme } from '~/theme/mui'
+import { setCurrencySelected } from '~/logic/currencyValues/store/actions/setCurrencySelected'
+import CheckIcon from './img/check.svg'
+import { AVAILABLE_CURRENCIES } from '~/logic/currencyValues/store/model/currencyValues'
+import fetchCurrencySelectedValue from '~/logic/currencyValues/store/actions/fetchCurrencySelectedValue'
+import { currentCurrencySelector } from '~/logic/currencyValues/store/selectors'
+
+const buttonWidth = '140px'
+const useStyles = makeStyles({
```

Comment:
We like to keep our components small `<200 LOC` with exceptions in rare cases. I think the styles in this component can be moved to a separate file so it's a little bit smaller

Reply:
done